subcommands/keys/rotate_root.go

```diff
@@ -276,3 +280,60 @@ func getOfflineCreds(credsFile string) OfflineCreds {
 	}
 	return files
 }
+
+func syncProdRoot(factory string, curRoot client.TufRoot, creds OfflineCreds) {
+	curProd, err := api.TufProdRootGet(factory)
+	if err != nil {
+		if httpE := client.AsHttpError(err); httpE != nil {
+			if httpE.Response.StatusCode == 404 {
+				// this is okay. it means we need to create the 1.rootjson
+				err = nil
+			}
+		}
+	}
+	subcommands.DieNotNil(err)
+	prodVer := 0
+	if curProd != nil {
+		prodVer = curProd.Signed.Version
+	}
+	if prodVer == curRoot.Signed.Version {
+		return
+	}
+
+	for i := prodVer + 1; i <= curRoot.Signed.Version; i++ {
```

Comment:
This doesn't seem good to me, by the time we do this old root keys might already be long gone.
Actually, I'd completely give up trying to keep a version in sync, and instead focus on keeping other fields in sync.
That allows to simplify an implementation a lot, as we only need to tackle these three use cases:
1. `root.json` and `prod-root.json` are the same except for a version and targets threshold - just update both of them.
2. `prod-root.json` is missing - just update `root.json` and generate `prod-root.json` (version 1)
3. `root.json` and `prod-root.json` are not the same, meaning there was an error updating `prod-root.json` - only update a `prod-root.json` and tell a user that we've just recovered from a sync error.

Also, I would disallow any modifications to prod.json on server side if a prod-root.json is out of sync.
This allows to keep implementation extremely simple and limit a space for errors.

Reply:
> This doesn't seem good to me, by the time we do this old root keys might already be long gone.

I don't understand what you mean here.

Reply:
What I mean is literally this: old root keys might not be present in a creds file, and in this case we'll not be able to backfill prod-root.json.
Assume this situation:
1. User rotates offline root keys, generates 1.root.json, there are keys root-0 and root-1 in creds
2. User rotates again and again, until reaching 10.root.json, there are now keys root-0 through root-10, if I understand a current logic.
But, only the last key is really in use, other keys are no more needed, and a user may decide to remove them from a creds file, that's a quite legitimate workflow and is actually good from security perspective.
I'd even say we need to cleanup old keys automatically.
3. Now we release prod targets and a user decides to rotate again.
    At this point only key root-10 is present, so we're not able to backfill all prod-root.json files.

And what I say is we don't need to backfill them, all we need is generate the first version of prod-root.json which is identical to root.json except for version, targets threshold, and signatures.